From d863599cc7986532ee4f05a7710a980739d541db Mon Sep 17 00:00:00 2001
Message-Id: <d863599cc7986532ee4f05a7710a980739d541db.1430272235.git.chang-joon.lee@intel.com>
In-Reply-To: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
References: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
From: vkanduri <venkatrajkumar.kanduri@intel.com>
Date: Fri, 17 Apr 2015 13:23:11 +0530
Subject: [PATCH 10/10] REVERTME [VPG]: drm/i915: Limit CHV CDCLK

CHV only supports cdclk upto 320MHz.
Any attempt to go higher results in an DRM error message.
In CHV cdclk 320MHz is enough for enabling 4kx2k@30Hz resolution.
This is because CHV can support pixel clock upto 95% VCO clock.
Hence limiting the max clock calculated to 320MHz to avoid the error.

REVERTME
Since upstream already has this implementation and next fork lift will
bring this change. So only putting this change for current requirements.

Issue: IMINAN-15948
Change-Id: I8b5701d288e57aa84594d96b1ad496bf89b9fb60
Signed-off-by: vkanduri <venkatrajkumar.kanduri@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d4692be..904d06a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5252,11 +5252,11 @@ static int valleyview_calc_cdclk(struct drm_i915_private *dev_priv,
 	 *   200MHz
 	 *   267MHz
 	 *   320MHz
-	 *   400MHz
+	 *   400MHz -- Not supported in CHV
 	 * So we check to see whether we're above 90% of the lower bin and
 	 * adjust if needed.
 	 */
-	if (max_pixclk > 288000)
+	if (max_pixclk > 288000 && !IS_CHERRYVIEW(dev_priv->dev))
 		new_cdclk = 400;
 	else if (max_pixclk > 240000)
 		new_cdclk = 320;
-- 
1.7.9.5

